src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionInitializer.java

```diff
@@ -130,34 +130,27 @@ public AlleleFractionInitializer(final AlleleFractionData data) {
     }
 
     private double estimateOutlierProbability(final AlleleFractionData data) {
-        final UnivariateObjectiveFunction objective = new UnivariateObjectiveFunction(outlierProbability ->
-                AlleleFractionLikelihoods.logLikelihood(state.shallowCopyWithProposedOutlierProbability(outlierProbability), data));
-        final SearchInterval searchInterval = new SearchInterval(0.0, MAX_REASONABLE_OUTLIER_PROBABILITY, state.outlierProbability());
-        return OPTIMIZER.optimize(objective, GoalType.MAXIMIZE, searchInterval, BRENT_MAX_EVAL).getPoint();
+        final Function<Double, Double> objective = outlierProbability ->
```

Comment:
So, my reasoning was that a shallow copy of the whole state is a lot scarier because of the potential for abuse, whereas copying an object that consists only of primitives is inherently fast and safe.  But I see the arguments against the refactoring as well of course.  I think that it ended up making the HMM code more natural, but I definitely want a second opinion.